src/renderers/shared/fiber/__tests__/ReactIncremental-test.js

```diff
@@ -217,4 +219,68 @@ describe('ReactIncremental', function() {
 
   });
 
+  it('can resume work in a subtree even when a parent bails out', function() {
+
+    var ops = [];
+
+    function Bar(props) {
+      ops.push('Bar');
+      return <div>{props.children}</div>;
+    }
+
+    function Tester() {
+      // This component is just here to ensure that the bail out is
+      // in fact in effect in the expected place for this test.
+      ops.push('Tester');
+      return <div />;
+    }
+
+    function Middle(props) {
+      ops.push('Middle');
+      return <span>{props.children}</span>;
+    }
+
+    var middleContent = (
+      <aaa>
+        <Tester />
+        <bbb hidden={true}>
+          <ccc>
+            <Middle>Hi</Middle>
+          </ccc>
+        </bbb>
+      </aaa>
+    );
+
+    function Foo(props) {
+      ops.push('Foo');
+      return (
+        <div>
+          <Bar>{props.text}</Bar>
+          {middleContent}
+          <Bar>{props.text}</Bar>
+        </div>
+      );
+    }
+
+    // Init
+    ReactNoop.render(<Foo text="foo" />);
+    ReactNoop.flushLowPri(52);
+
+    expect(ops).toEqual(['Foo', 'Bar', 'Tester', 'Bar']);
+
+    ops = [];
+
+    // We're now rendering an update that will bail out on updating middle.
+    ReactNoop.render(<Foo text="bar" />);
+    ReactNoop.flushLowPri(45);
+
+    expect(ops).toEqual(['Foo', 'Bar', 'Bar']);
+
+    ops = [];
+
+    // Flush the rest to make sure that the bailout didn't block this work.
+    ReactNoop.flush();
+    expect(ops).toEqual(['Middle']);
```

Comment:
Middle doesn't actually update. It is the initial mount. Note that it has been down-prioritized in the initial mount. The initial mount finished but for anything with the default low priority but didn't finish the offscreen priority.

So this last piece is actually just finishing up existing work that didn't already finish. It's not an update. Where as Tester already finished earlier.